### DISCUSSION TITLE
Browser not loading on  attached LCD panel with 20230918

### DISCUSSION THREAD

**Discussion by cYCL157** [rern/rAudio#1435] [Bug Reports]
I am having an issue since updating to 20230918 where the directly attached pi panel does not show browser but a login prompt.  I have tried to play around with things a bit but not quite sure where to dig in here.  I am running rAudio 20230918 on a Raspberry Pi 3B+ Rev 1.3 with HiFiBerry DAC+ and a Raspberry Pi 7" touch screen display.  I have been running on this hardware for almost 3 years and have not had issues until now.  

I really appreciate any suggestions on what I can do or what more I might be able to provide to make this report of value.

Here is the output from the localbrowser feature:

firefox 117.0.1-1
# cat /srv/http/data/system/localbrowser.conf
rotate=0
zoom=70
screenoff=0
onwhileplay=
hdmi=
cursor=

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; disabled; vendor preset: disabled)
     Active: inactive (dead)

Sep 22 18:47:17 rAudio sudo[126242]: grep: /root/.mozilla/firefox/profiles.ini: No such file or directory
Sep 22 18:47:17 rAudio sudo[126236]: matchbox-window-manager: Failed to connect to session manager
Sep 22 18:47:17 rAudio sudo[126223]: /etc/X11/xinit/xinitrc: line 29: /root/.mozilla/firefox//user.js: No such file or directory
Sep 22 18:47:17 rAudio sudo[126244]: /usr/lib/firefox/firefox: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/firefox/firefox)
Sep 22 18:47:17 rAudio sudo[126218]: xinit: connection to X server lost
Sep 22 18:47:17 rAudio sudo[126218]: 
Sep 22 18:47:17 rAudio sudo[126218]: waiting for X server to shut down
Sep 22 18:47:17 rAudio sudo[126219]: (II) Server terminated successfully (0). Closing log file.
Sep 22 18:47:18 rAudio sudo[126201]: pam_unix(sudo:session): session closed for user root
Sep 22 18:47:18 rAudio systemd[1]: localbrowser.service: Deactivated successfully.

**Comment by ceejayemm** [comment]
Have you tried disabling and re-enabling Features / Others / Browser on RPi ? Worth a try?  I am running this version with the official RPi 7in touchscreen with no problems (except setting the Zoom level #1429) since Firefox was implemented.

Chris

**Reply by cYCL157** [reply]
i have, a few times, including rebooting.  the display is on, just shows login prompt.  I can't say that I looked when I updated to 20230917, so I don't know which release it might be associated with -- although 0917 did have firefox in it.

When I try I get the following in the logs:

Sep 28 18:22:05 rAudio systemd[1]: Started Browser on RPi.
-- Subject: A start job for unit localbrowser.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit localbrowser.service has finished successfully.
--
-- The job identifier is 2259.
Sep 28 18:22:05 rAudio ply-image[346221]: [25B blob data]
Sep 28 18:22:05 rAudio systemd[1]: Finished Splash screen on startup.
-- Subject: A start job for unit bootsplash.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit bootsplash.service has finished successfully.
--
-- The job identifier is 2256.
Sep 28 18:22:05 rAudio audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=bootsplash comm="systemd" exe="/usr/lib/systemd/systemd" h>
Sep 28 18:22:05 rAudio audit[346222]: USER_ACCT pid=346222 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix,pam_permit,pam_time a>
Sep 28 18:22:05 rAudio sudo[346222]:     root : PWD=/ ; USER=root ; COMMAND=/usr/bin/startx
Sep 28 18:22:05 rAudio audit[346222]: CRED_REFR pid=346222 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,p>
Sep 28 18:22:05 rAudio audit[346222]: USER_START pid=346222 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_limits,pam_unix,pam_per>
Sep 28 18:22:05 rAudio sudo[346222]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Sep 28 18:22:06 rAudio sudo[346240]: X.Org X Server 1.21.1.3
Sep 28 18:22:06 rAudio sudo[346240]: X Protocol Version 11, Revision 0
Sep 28 18:22:06 rAudio sudo[346240]: Current Operating System: Linux rAudio 5.15.35-1-rpi-ARCH #1 SMP PREEMPT Tue Apr 26 17:24:19 UTC 2022 aarch64
Sep 28 18:22:06 rAudio sudo[346240]: Kernel command line: coherent_pool=1M 8250.nr_uarts=0 snd_bcm2835.enable_compat_alsa=0 snd_bcm2835.enable_hdmi=1 bcm2708_>
Sep 28 18:22:06 rAudio sudo[346240]:
Sep 28 18:22:06 rAudio sudo[346240]: Current version of pixman: 0.40.0
Sep 28 18:22:06 rAudio sudo[346240]:         Before reporting problems, check http://wiki.x.org
Sep 28 18:22:06 rAudio sudo[346240]:         to make sure that you have the latest version.
Sep 28 18:22:06 rAudio sudo[346240]: Markers: (--) probed, (**) from config file, (==) default setting,
Sep 28 18:22:06 rAudio sudo[346240]:         (++) from command line, (!!) notice, (II) informational,
Sep 28 18:22:06 rAudio sudo[346240]:         (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
Sep 28 18:22:06 rAudio sudo[346240]: (==) Log file: "/var/log/Xorg.0.log", Time: Thu Sep 28 18:22:06 2023
Sep 28 18:22:06 rAudio sudo[346240]: (==) Using config directory: "/etc/X11/xorg.conf.d"
Sep 28 18:22:06 rAudio sudo[346240]: (==) Using system config directory "/usr/share/X11/xorg.conf.d"
Sep 28 18:22:06 rAudio sudo[346242]: modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/5.15.35-1-rpi-ARCH
Sep 28 18:22:07 rAudio systemd[1]: Reloading.
Sep 28 18:22:08 rAudio audit: BPF prog-id=303 op=LOAD
Sep 28 18:22:08 rAudio kernel: kauditd_printk_skb: 37 callbacks suppressed
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.575:3694): prog-id=303 op=LOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.575:3695): prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.579:3696): prog-id=304 op=LOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.579:3697): prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.583:3698): prog-id=305 op=LOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.583:3699): prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.587:3700): prog-id=306 op=LOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.587:3701): prog-id=307 op=LOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.587:3702): prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio kernel: audit: type=1334 audit(1695946928.587:3703): prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=304 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=305 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=306 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=307 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=308 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=309 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=310 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=311 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=312 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=313 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=314 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=315 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=316 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=317 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=318 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=319 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=320 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=321 op=LOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio audit: BPF prog-id=0 op=UNLOAD
Sep 28 18:22:08 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Sep 28 18:22:08 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Sep 28 18:22:10 rAudio sudo[346168]: pam_unix(sudo:session): session closed for user root
Sep 28 18:22:10 rAudio audit[346168]: USER_END pid=346168 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_limits,pam_unix,pam_per>
Sep 28 18:22:10 rAudio audit[346168]: CRED_DISP pid=346168 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,>
Sep 28 18:22:10 rAudio sudo[346380]: The XKEYBOARD keymap compiler (xkbcomp) reports:
Sep 28 18:22:10 rAudio sudo[346380]: > Warning:          Could not resolve keysym XF86EmojiPicker
Sep 28 18:22:10 rAudio sudo[346380]: Errors from xkbcomp are not fatal to the X server
Sep 28 18:22:10 rAudio sudo[346384]: Failed to connect to bus: No medium found
Sep 28 18:22:10 rAudio sudo[346385]: dbus-update-activation-environment: error: unable to connect to D-Bus: Using X11 for dbus-daemon autolaunch was disabled >
Sep 28 18:22:10 rAudio sudo[346401]: grep: /root/.mozilla/firefox/profiles.ini: No such file or directory
Sep 28 18:22:10 rAudio sudo[346382]: /etc/X11/xinit/xinitrc: line 29: /root/.mozilla/firefox//user.js: No such file or directory
Sep 28 18:22:10 rAudio sudo[346403]: /usr/lib/firefox/firefox: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/firefox/firef>
Sep 28 18:22:10 rAudio sudo[346239]: xinit: connection to X server lost
Sep 28 18:22:10 rAudio sudo[346239]:
Sep 28 18:22:10 rAudio sudo[346239]: waiting for X server to shut down
Sep 28 18:22:10 rAudio sudo[346240]: (II) Server terminated successfully (0). Closing log file.
Sep 28 18:22:11 rAudio sudo[346222]: pam_unix(sudo:session): session closed for user root
Sep 28 18:22:11 rAudio audit[346222]: USER_END pid=346222 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_limits,pam_unix,pam_perm>
Sep 28 18:22:11 rAudio audit[346222]: CRED_DISP pid=346222 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,p>
Sep 28 18:22:11 rAudio audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=localbrowser comm="systemd" exe="/usr/lib/systemd/systemd" >
Sep 28 18:22:11 rAudio systemd[1]: localbrowser.service: Deactivated successfully.
-- Subject: Unit succeeded
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- The unit localbrowser.service has successfully entered the 'dead' state.
Sep 28 18:22:17 rAudio mpd[787]: player: played "NAS/Pandora/80s Rock Ballads Radio/Phil Collins/No Jacket Required (2016 Remaster)/Phil Collins - Sussudio (2>
Sep 28 18:25:52 rAudio audit[346552]: USER_ACCT pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix,pam_permit,pam_time >
Sep 28 18:25:52 rAudio kernel: kauditd_printk_skb: 33 callbacks suppressed
Sep 28 18:25:52 rAudio kernel: audit: type=1101 audit(1695947152.317:3737): pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=p>
Sep 28 18:25:52 rAudio kernel: audit: type=1110 audit(1695947152.329:3738): pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_>
Sep 28 18:25:52 rAudio kernel: audit: type=1105 audit(1695947152.329:3739): pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors>
Sep 28 18:25:52 rAudio audit[346552]: CRED_REFR pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,>
Sep 28 18:25:52 rAudio audit[346552]: USER_START pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_limits,pam_unix,pam_pe>
Sep 28 18:25:52 rAudio sudo[346552]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/features-data.sh
Sep 28 18:25:52 rAudio sudo[346552]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Sep 28 18:25:52 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Sep 28 18:25:52 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Sep 28 18:25:52 rAudio sudo[346552]: pam_unix(sudo:session): session closed for user root
Sep 28 18:25:52 rAudio audit[346552]: USER_END pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_limits,pam_unix,pam_per>
Sep 28 18:25:52 rAudio audit[346552]: CRED_DISP pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,>
Sep 28 18:25:52 rAudio kernel: audit: type=1106 audit(1695947152.769:3740): pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantor>
Sep 28 18:25:52 rAudio kernel: audit: type=1104 audit(1695947152.769:3741): pid=346552 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_>
lines 505-552/552 (END)

**Comment by cYCL157** [comment]
Still having this issue after update to 20231001.

**Comment by cYCL157** [comment]
Still having issues with 202310131.  Here is the output from the information on the option:

 cat /srv/http/data/system/localbrowser.conf
rotate=0
zoom=70
screenoff=0
onwhileplay=
hdmi=
cursor=

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; enabled; vendor preset: disabled)
     Active: inactive (dead) since Sun 2023-10-15 09:00:44 MDT; 24s ago
    Process: 59654 ExecStart=/usr/bin/sudo /usr/bin/startx (code=exited, status=0/SUCCESS)
   Main PID: 59654 (code=exited, status=0/SUCCESS)
        CPU: 740ms

Oct 15 09:00:43 rAudio sudo[59698]: dbus-update-activation-environment: error: unable to connect to D-Bus: Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead
Oct 15 09:00:43 rAudio sudo[59714]: grep: /root/.mozilla/firefox/profiles.ini: No such file or directory
Oct 15 09:00:43 rAudio sudo[59695]: /etc/X11/xinit/xinitrc: line 29: /root/.mozilla/firefox//user.js: No such file or directory
Oct 15 09:00:43 rAudio sudo[59716]: /usr/lib/firefox/firefox: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/firefox/firefox)
Oct 15 09:00:43 rAudio sudo[59671]: xinit: connection to X server lost
Oct 15 09:00:43 rAudio sudo[59671]: 
Oct 15 09:00:43 rAudio sudo[59671]: waiting for X server to shut down
Oct 15 09:00:43 rAudio sudo[59672]: (II) Server terminated successfully (0). Closing log file.
Oct 15 09:00:44 rAudio sudo[59654]: pam_unix(sudo:session): session closed for user root
Oct 15 09:00:44 rAudio systemd[1]: localbrowser.service: Deactivated successfully.

**Comment by cYCL157** [comment]
Still a problem with 20231020.  Attempt to start reports it fails:

firefox 117.0.1-1
# cat /srv/http/data/system/localbrowser.conf
rotate=0
zoom=70
screenoff=0
onwhileplay=
cursor=
runxinitrcd=

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; disabled; vendor preset: disabled)
     Active: inactive (dead)

Oct 20 11:23:44 rAudio sudo[2348]: Errors from xkbcomp are not fatal to the X server
Oct 20 11:23:44 rAudio sudo[2363]: ls: cannot access '/root/.mozilla/firefox': No such file or directory
Oct 20 11:23:44 rAudio sudo[2350]: /etc/X11/xinit/xinitrc: line 21: /root/.mozilla/firefox//user.js: No such file or directory
Oct 20 11:23:44 rAudio sudo[2365]: /usr/lib/firefox/firefox: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/firefox/firefox)
Oct 20 11:23:44 rAudio sudo[2345]: xinit: connection to X server lost
Oct 20 11:23:44 rAudio sudo[2345]: 
Oct 20 11:23:44 rAudio sudo[2345]: waiting for X server to shut down
Oct 20 11:23:44 rAudio sudo[2346]: (II) Server terminated successfully (0). Closing log file.
Oct 20 11:23:45 rAudio sudo[2328]: pam_unix(sudo:session): session closed for user root
Oct 20 11:23:45 rAudio systemd[1]: localbrowser.service: Deactivated successfully.

**Comment by rern** [comment]
Have you done the `pacman -Syu`?

Kernels newer than 6.1.54 - 6.1.57 break the browser on local display, something about X server. If it's the case, `pacman -Syu` again to get the kernel upgraded to 6.1.58. Just tested and it runs fine now.

**Reply by cYCL157** [reply]
i have not, but will -- here is the version I have now:

rAudio:~  #  uname -srm
Linux 5.15.35-1-rpi-ARCH aarch64

After running pacman -Syu, I get the following.....any advice?

 Total (503/503)                                                    615.4 MiB  2.96 MiB/s 03:28 [########################################################] 100%
(503/503) checking keys in keyring                                                              [########################################################] 100%
(503/503) checking package integrity                                                            [########################################################] 100%
(503/503) loading package files                                                                 [########################################################] 100%
(503/503) checking for file conflicts                                                           [########################################################] 100%
error: failed to commit transaction (conflicting files)
camilladsp: /etc/default/camilladsp exists in filesystem
Errors occurred, no packages were upgraded.

**Reply by cYCL157** [reply]
the directory was camilladsp directory was empty so I just removed it and pacman finished.  I am not getting the remote UI to work and trying to figure out why.  it is playing music from my nas server -- I plugged it in hardwire just in case it was something funky with wireless and no dice.  I'll flash a new card.

**Comment by ceejayemm** [comment]
If your kernel is 5.15 (mine is 6.1.58-2) it would seem you are running a very old rAudio base. I am using i20230918 updated to 20231020 on a Pi3B+ with the official touchscreen and now (after applying packman-Syu which i wouldn't normally do) have few problems with this setup, certainly not like you have. 

Chris

**Reply by cYCL157** [reply]
Thanks.  I haven't ever run pacman as part of updates, maybe I missed that I should be and until recently haven't had any problems with things being updated.  If pacman updates are are suggested or needed, I wonder if it would be better to just make that part of the update.  I don't envy @rern  trying to support multiple versions of hardware and kernel versions.....that seems daunting!  that said, his efforts and those of anyone helping is much appreciated, I really enjoy rAudio!

**Reply by ceejayemm** [reply]
I don't normally run pacman either as I think @rern  does a good job with the main releases. It was only because rern suggested it in #1435 that I did it on this occasion

Chris

**Comment by cYCL157** [comment]
alright -- i seem to be back up again.  i didn't have a recent backup so am having to rebuild my library again.  Kernel version is 6.1.53 from the flashed card from 20230918.  I will let the library build again, get a backup and then run pacman again to update to current and confirm things still work.  Hopefully that confirms things and is helpful for someone else.

Thanks @ceejayemm and @rern !!
